### PR TITLE
feat: AppDataUpdateData intent serialization proto

### PR DESCRIPTION
**Pull request by tylerhawkes**
## Summary

Adds `AppDataUpdateData` + `AppDataUpdateDeltaWithBase` to `xmtp.mls.database` for local-only intent serialization of libxmtp's new generic `IntentKind::AppDataUpdate` (https://github.com/xmtp/libxmtp/pull/3664).

The proto carries:
- `component_id: uint32` (libxmtp's `u16` widened — proto has no u16)
- `oneof op` with:
  - `bytes replace` — full-replace components (Bytes/String typed)
  - `AppDataUpdateDeltaWithBase delta_with_base` — collection components (TlsMap/TlsSet) with `pre_value`+`post_value` for 3-way merge

Never crosses the MLS wire; stored only in libxmtp's local SQLite `group_intents` table.

## Test plan

- [x] Pinned at SHA `7b8260c3041389bf706fb665460adbbd520d2be7`.
- [x] libxmtp regen succeeds with `GEN_PROTOS=true cargo check -p xmtp_proto`.
- [x] Round-trip + edge-case tests in libxmtp PR #3664 pass.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

> [!NOTE]
> ### Add `AppDataUpdateData` protobuf message for app data update intent serialization
> Adds a new versioned protobuf message `AppDataUpdateData` in [intents.proto](https://github.com/xmtp/proto/pull/333/files#diff-6f0fe39046ec77adf0be4b95d56a75c664665c04f7c30965c66d2e5cd24319f0) with a `V1` envelope containing `component_id` (uint32) and `payload` (bytes). Inline comments document the expected payload semantics for `Bytes`/`String`, `TlsMap`, and `TlsSet` component types.
>
> 
>
> <sup><a href="https://app.macroscope.com">Macroscope</a> summarized de79332.</sup>
> 
>

### CODE REVIEW

File: proto/mls/database/intents.proto [hunk 1]
```diff
@@ -165,6 +165,45 @@ message UpdatePermissionData {
...
+    // time.
+    AppDataUpdateDeltaWithBase delta_with_base = 3;
+  }
+}
+
+message AppDataUpdateDeltaWithBase {
+  // Full prior bytes of the component as observed when the caller
+  // built the intent. Merge base for the 3-way merge.
+  bytes pre_value = 1;
+  // Full desired post bytes of the component.
```
Comment (neekolas) [inline proto/mls/database/intents.proto]: Do you have some reference on how we resolve the residual mutation and do the 3-way-merge?

I like this approach in general, just want to learn more about the process since it seems important to grok moving forward.

**Review by neekolas — APPROVED** [review]

File: proto/mls/database/intents.proto (hunk 1 above)
Comment (tylerhawkes) [reply proto/mls/database/intents.proto]: For components that are Maps/Sets you can just write out the current state and the desired state and when doing the commit we can figure out what the delta is to get into the desired state. So if Alix wants to add inboxes x and y, but Bo adds X in the meantime, we can still figure out that we need to add y and just make that change. I'll have to do more tests with interim removals, but the idea is that we can state the end state (either total, or the diffs) and then choose to not publish some of the changes if they've already been made.

Comment (tylerhawkes) [reply proto/mls/database/intents.proto]: After thinking about it, we're going to go with last writer wins for now. We can update this later if we want some better semantics around races, but those should be rare enough that we maybe don't care about them.
